packages/@expo/cli/src/start/server/metro/resolveFromProject.ts

```diff
@@ -44,9 +44,10 @@ export function importMetroHmrServerFromProject(
 }
 
 export function importExpoMetroConfig(projectRoot: string) {
-  return importFromProjectOrFallback<typeof import('@expo/metro-config')>(
+  // `@expo/metro-config` version is controlled by `@expo/cli` to avoid conflicting `@expo/metro-config` versions in project dependencies
+  return importFromProjectOrFallback<typeof import('@expo/cli/metro-config')>(
     projectRoot,
-    '@expo/metro-config'
+    '@expo/cli/metro-config'
```

Comment:
This doesn’t seem like it needs to change

Reply:
I thought it didn't either, but the reason why these also need to change it because they resolve from `projectRoot`. Basically, we treat all modules here as being a "list" inside `<project>/node_modules`, while technically we should treat it as a proper dependency graph.

In this case, what happens is `@expo/metro-config` is resolved from project root, or in other words `<project>/node_modules>`. Meaning that if the user, or any other library, added the incorrect `@expo/metro-config@0.0.1` version, we are still pulling in the wrong one.

The correct way of resolving this, and other libraries in this file, like how we would traverse dependency graphs. E.g.
- resolve `expo`
- use `expo` location to resolve `@expo/cli`
- use `@expo/cli` location to resolve `@expo/metro-config`